src/SFA.DAS.EmploymentCheck.Functions/Application/Services/Hmrc/HmrcService.cs

```diff
@@ -199,63 +153,5 @@ private async Task RetrieveAuthenticationToken()
-        {
-            if (employmentCheckCacheResponse == null) return await Task.FromResult(0);
-
-            var dbConnection = new DbConnection();
-            await using (var sqlConnection = await dbConnection.CreateSqlConnection(
-                _connectionString,
-                AzureResource,
-                _azureServiceTokenProvider))
-            {
-                Guard.Against.Null(sqlConnection, nameof(sqlConnection));
-
-                // There is a constraint to stop duplicates but this check avoids the exception causing a problem later in the code
-                await sqlConnection.OpenAsync();
-                var parameter = new DynamicParameters();
-                parameter.Add("@existingApprenticeEmploymentCheckId", employmentCheckCacheResponse.ApprenticeEmploymentCheckId, DbType.Int64);
-                parameter.Add("@existingEmploymentCheckCacheRequestId", employmentCheckCacheResponse.EmploymentCheckCacheRequestId, DbType.Int64);
-                parameter.Add("@existingCorrelationId", employmentCheckCacheResponse.CorrelationId, DbType.Guid);
-                parameter.Add("@existingFoundOnPaye", employmentCheckCacheResponse.FoundOnPaye, DbType.String);
-
-                parameter.Add("@apprenticeEmploymentCheckId", employmentCheckCacheResponse.ApprenticeEmploymentCheckId, DbType.Int64);
-                parameter.Add("@employmentCheckCacheRequestId", employmentCheckCacheResponse.EmploymentCheckCacheRequestId, DbType.Int64);
-                parameter.Add("@correlationId", employmentCheckCacheResponse.CorrelationId, DbType.Guid);
-                parameter.Add("@employed", employmentCheckCacheResponse.Employed, DbType.Boolean);
-                parameter.Add("@foundOnPaye", employmentCheckCacheResponse.FoundOnPaye, DbType.String);
-                parameter.Add("@processingComplete", employmentCheckCacheResponse.ProcessingComplete, DbType.Boolean);
-                parameter.Add("@count", employmentCheckCacheResponse.Count, DbType.Int32);
-                parameter.Add("@httpResponse", employmentCheckCacheResponse.HttpResponse, DbType.String);
-                parameter.Add("@httpStatusCode", employmentCheckCacheResponse.HttpStatusCode, DbType.Int16);
-                parameter.Add("@createdOn", DateTime.Now, DbType.DateTime);
-
-                await sqlConnection.ExecuteScalarAsync(
-                "IF NOT EXISTS " +
-                "( " +
-                "   SELECT  [ApprenticeEmploymentCheckId] " +
-                "   FROM    [Cache].[EmploymentCheckCacheResponse] " +
-                "   WHERE   [ApprenticeEmploymentCheckId] = @existingApprenticeEmploymentCheckId " +
-                "   AND     [EmploymentCheckCacheRequestId] = @existingEmploymentCheckCacheRequestId " +
-                "   AND     [CorrelationId] = @existingCorrelationId " +
-                "   AND     [FoundOnPaye] = @existingFoundOnPaye " +
-                ") " +
```

Comment:
Sorry Rus, I'm not that familiar with Dapper Contrib. I've just put the 'IF NOT EXISTS' check in as part of the fix for 182 so that we don't get the exception from the constraint on the table taking us out of the method and skipping the rest of the rows on the batch. Does the repository do an existence check if the row already exists before doing the insert so that we don't get an exception from the constraint on the table?